packages/neon_framework/lib/src/login_flow_v2/view/login_flow_v2_view.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:meta/meta.dart';
+import 'package:neon_framework/l10n/localizations.dart';
+import 'package:neon_framework/src/login/login.dart';
+import 'package:neon_framework/src/login_flow_v2/login_flow_v2.dart';
+
+@internal
+class LoginFLowV2View extends StatelessWidget {
+  const LoginFLowV2View({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocConsumer<LoginFlowV2Bloc, LoginFlowV2State>(
+      listener: (context, state) {
+        final credentials = state.credentials;
+        if (credentials != null) {
+          context.read<LoginBloc>().add(LoginCredentialsEntered(credentials));
+        }
+      },
```

Comment:
You are right to some degree.
This part does not belong here, but it must still be in a widget. Two blocs should never know of one another to avoid interlocked blocks.
The optimal place would be in the page so other views can reuse it.

Unfortunately it is not easy to unit test as we can not mock the `LoginFlowV2Bloc` in the page test. In an ideal case the `LoginFlowV2Bloc` would be inserted by the route (MaterialPageRoute), the listener would sit in the page and the view would just render the state.

Due to us using go_router this isn't a possibility now and would scatter the code arround. I'll add a todo comment to change it once we refactored our routing but this will take some time and the current solution is a tradeoff I'm willing to make. wdyt?

Reply:
Sounds like a good compromise